kytos-amlight-topo-init.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+KYTOS_URL="http://127.0.0.1:8181/api"
+TOPOLOGY_API="kytos/topology/v3"
+LLDP_API="kytos/of_lldp/v1"
+
+http_post() {
+	URL_PATH="$1"
+	BODY="$2"
+
+	echo "--> POST $URL_PATH $BODY" 1>&2
+	curl -H 'Content-Type: application/json' -X POST $KYTOS_URL/$URL_PATH -d "$BODY"
+}
+
+http_get() {
+	URL_PATH="$1"
+	echo "--> GET $URL_PATH" 1>&2
+	curl -s $KYTOS_URL/$URL_PATH
+}
+
+set_switch_enable_metadata() {
+	DPID="$1"
+	DATA="$2"
+
+	if [ $(echo -n $DPID | wc -c) -eq 2 ]; then
+		DPID=00:00:00:00:00:00:00:$DPID
+	fi
+
+	http_post $TOPOLOGY_API/switches/$DPID/enable
+	http_post $TOPOLOGY_API/switches/$DPID/metadata "$DATA"
+}
+
+disable_lldp_all_interfaces() {
+	ALL_INTERFACES=$(http_get $LLDP_API/interfaces)
+
+	http_post $LLDP_API/interfaces/disable "$ALL_INTERFACES"
+}
+
+set_interface_enable_metadata() {
+	IFACEID="$1"
+	DATA="$2"
+
+	http_post $TOPOLOGY_API/interfaces/$IFACEID/enable
+	http_post $TOPOLOGY_API/interfaces/$IFACEID/metadata "$DATA"
+}
+
+set_interface_enable_lldp_metadata() {
+	IFACEID="$1"
+	DATA="$2"
+
+	http_post $TOPOLOGY_API/interfaces/$IFACEID/enable
+	http_post $TOPOLOGY_API/interfaces/$IFACEID/metadata "$DATA"
+	http_post $LLDP_API/interfaces/enable "{\"interfaces\": [\"$IFACEID\"]}"
+}
+
+set_link_enable_metadata() {
+	LINKID="$1"
+	DATA="$2"
+
+	http_post $TOPOLOGY_API/links/$LINKID/enable
+	http_post $TOPOLOGY_API/links/$LINKID/metadata "$DATA"
+}
+
+wait_on() {
+	TYPE=$1
+	NUMBER=$2
+
+	i=0
+	while [ $i -lt 300 ]; do
+		if [ $(curl -s $KYTOS_URL/$TOPOLOGY_API/$TYPE | jq -r ".$TYPE[].id" | wc -l) -eq $NUMBER ]; then
+			break
+		fi
+		sleep 1
+		i=$(($i+1))
+	done
+	if [ $i -eq 300 ]; then
+		echo "ERROR: missing $TYPE. Current $TYPE:"
+		curl -s $KYTOS_URL/$TOPOLOGY_API/$TYPE | jq -r ".$TYPE[].id"
+		exit 1
+	fi
+}
+
+#############################
+# Setup switches
+#############################
+wait_on switches 12
```

Comment:
I've updated it from 11 to 12 since there are 12 switches:

```
❯ http localhost:8181/api/kytos/topology/v3/switches | jq -r '.switches[].id' | wc -l                                                                                          
12
```

Reply:
Excellent, thanks @viniarck !